src/Core/Domain/Product/QueryResult/ProductCustomizability.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ProductCustomizabilitySettings;
+
+/**
+ * Transfers data of product customizability value
+ */
+class ProductCustomizability
+{
+    /**
+     * @var int
+     */
+    private $value;
+
+    /**
+     * @var int
+     */
+    private $availableTextCustomizations;
+
+    /**
+     * @var int
+     */
+    private $availableFileCustomizations;
+
+    /**
+     * @return ProductCustomizability
+     */
+    public static function createNotCustomizable(): ProductCustomizability
+    {
+        return new self(
+            ProductCustomizabilitySettings::NOT_CUSTOMIZABLE,
+            0,
+            0
+        );
+    }
+
+    /**
+     * @param int $availableTextCustomizations
+     * @param int $availableFileCustomizations
+     *
+     * @return ProductCustomizability
+     */
+    public static function createAllowsCustomization(
+        int $availableTextCustomizations,
```

Comment:
```suggestion
        int $numberOfAllowedTextCustomizations,
```
your wording was good but maybe we can make it even more unambiguous 😉 

When possible, imagine you read your code to a colleague and in an ideal world, the colleague would understand what it does because it sounds like English.

## Before

"I create an object that allows customization with five available text customizations and two available file customizations"

## After

"I create an object that allows customization with a number of allowed text customizations of five and a number of allowed file customizations of two.